webapp/src/routes/IncomeReport/IncomeTable.js

```diff
@@ -1,99 +1,127 @@
-import React, { memo } from 'react'
+import React, { memo, useState } from 'react'
 import PropTypes from 'prop-types'
 import { useTranslation } from 'react-i18next'
+import { DataGrid, esES, enUS } from '@mui/x-data-grid'
+import { createTheme, ThemeProvider } from '@mui/material/styles'
 import { makeStyles } from '@mui/styles'
-import {
-  TableContainer,
-  Table,
-  TableHead,
-  TableRow,
-  TableCell,
-  TableBody,
-  Tooltip
-} from '@mui/material'
+import { Tooltip } from '@mui/material'
 
 import styles from './styles'
 
 const useStyles = makeStyles(styles)
+const rowsCenter = { flex: 1, align: 'center', headerAlign: 'center' }
 
 const IncomeTable = ({ data, thousandSeparator, dataPercent }) => {
+  const [pagePaginationSize, setPagePaginationSize] = useState(5)
+  const newDataTable = data.map(firstObj => ({
+    ...dataPercent.find(secondObj => secondObj.name === firstObj.name),
+    ...firstObj
+  }))
   const classes = useStyles()
-
   const { t } = useTranslation('incomeRoute')
+  const theme = createTheme(t('tableHeader1') === 'Name' ? enUS : esES)
+  const columns = [
+    {
+      field: 'txId',
+      headerName: t('tableHeader2'),
+      hide: !data[0]?.txId,
+      cellClassName: classes.chartLinks,
+      renderCell: param => (
+        <Tooltip title={param.value}>
+          <a
+            href={
+              param.value.lenght > 60
+                ? `https://bloks.io/transaction/${param.value}`
+                : `https://bloks.io/account/genesis.eden?loadContract=true&tab=Tables&table=distaccount&account=genesis.eden&scope=&limit=100&lower_bound=${param.value}&upper_bound=${param.value}`
```

Comment:
is this the best place to set these URL values?

Reply:
This kind of table only allow to set the URL in this place